### PR TITLE
Only register respawn callbacks if respawn is enabled

### PR DESCRIPTION
**Pull request by adrido**
Its only a very small optimization, but it is one.

I hope I followed the code style. If not, please complain :smile:

### CODE REVIEW

File: mods/beds/functions.lua [hunk 1]
```diff
@@ -173,23 +173,23 @@ end
 
 
 -- Callbacks
+-- Only register respawn callbacks if respawn enabled
+if enable_respawn then 
+	minetest.register_on_joinplayer(function(player)
+		beds.read_spawns()
+	end)
```
Comment (SmallJoker) [inline mods/beds/functions.lua]: #1102 removes this callback.

File: mods/beds/functions.lua [hunk 2]
```diff
@@ -173,23 +173,23 @@ end
...
+	
+	-- respawn player at bed if enabled and valid position is found
+	minetest.register_on_respawnplayer(function(player)
+		local name = player:get_player_name()
+		local pos = beds.spawn[name] or nil
```
Comment (SmallJoker) [inline mods/beds/functions.lua]: `or nil` doesn't make sense - why is it this way in the old code?

File: mods/beds/functions.lua (hunk 1 above)
Comment (paramat) [inline mods/beds/functions.lua]: #1102 is merged.

**Comment by rubenwardy** [comment]
Use git pull upstream master --rebase to make it so your PR is make on top of new changes.
You'll need to undo the two commits not by you, and then run the above command

```
git reset --hard 744b5e9
git log
# check that  	Only register respawn callbacks if respawn is enabled   is newest commit
git pull upstream master --rebase
# where upstream is the name of the remote pointing to minetest/minetest_game
# if it does not exist, run:
# git remote add upstream https://github.com/minetest/minetest_game.git

# Now it will tell you about merge conflicts
# Go to the file it mentions, and look for <<<<<<   and >>>>>>
# Fix the file so it is correct

git add .
git commit

git log
# check that the log is correct
# Should be the latest MTG log + your commit at the top

git push origin patch-1 -f
# where origin is the name of your github fork
```

**Comment by adrido** [comment]
Thanks a lot rubenwardy :sweat_smile: 

but now I have another trouble, Im currently at the step git commit and the following colourful editor opens. 
git_editor
How is this program called? It does not show any name so I cant google how it works.
How do I write the commit message? WASD seems to control the blinking cursor.
How do I tell the program that I'm finished and submit the commit?
:confused:

**Comment by rubenwardy** [comment]
That is VIM. You need to type :wq

**Comment by rubenwardy** [comment]
Did you scroll down in that pic? There should be a commit message above all those lines starting with hashes

**Comment by adrido** [comment]
nope :confused: but I think I was successful to reset all to the state as it was before. Ill try to rebase now again.

**Comment by adrido** [comment]
Ha, I think its working now :laughing:  thanks a lot again rubenwardy

**Comment by rubenwardy** [comment]
Yay :D 

Welcome to git

**Comment by adrido** [comment]
The next time I do it like this: https://xkcd.com/1597/
Seems to be much simpler. :laughing:

**Comment by paramat** [comment]
:+1:

**Comment by sofar** [comment]
:+1: assuming it's tested - trivial IMHO

**Comment by paramat** [comment]
d89bb69a22e7abe90c80e827e07ea20a2c633a35

**Pull request closed without merging**